code/modules/mob/living/basic/ruin_defender/mimic/mimic.dm

```diff
@@ -0,0 +1,381 @@
+
+/datum/action/innate/mimic
+	background_icon_state = "bg_default"
+	overlay_icon_state = "bg_default_border"
+
+/datum/action/innate/mimic/lock
+	name = "Lock/Unlock"
+	desc = "Toggle preventing yourself from being opened or closed."
+	button_icon = 'icons/hud/radial.dmi'
+	button_icon_state = "radial_lock"
+
+/datum/action/innate/mimic/lock/Activate()
+	var/mob/living/basic/mimic/crate/mimic = owner
+	mimic.locked = !mimic.locked
+	if(!mimic.locked)
+		to_chat(mimic, span_warning("You loosen up, allowing yourself to be opened and closed."))
+	else
+		to_chat(mimic, span_warning("You stiffen up, preventing anyone from opening or closing you."))
+
+// ****************************
+// COPYING (actually imitates target object) MIMIC
+// ****************************
+
+/mob/living/basic/mimic/copy
+	health = 100
+	maxHealth = 100
+	mob_biotypes = MOB_SPECIAL
+	ai_controller = /datum/ai_controller/basic_controller/mimic_copy
+	/// our creator
+	var/mob/living/creator = null
+	/// googly eyes overlay
+	var/static/mutable_appearance/googly_eyes = mutable_appearance('icons/mob/simple/mob.dmi', "googly_eyes")
+	/// do we overlay googly eyes over whatever we copy
+	var/overlay_googly_eyes = TRUE
+	/// do we take damage when we are not sentient and have no target
+	var/idledamage = TRUE
+
+/mob/living/basic/mimic/copy/Initialize(mapload, obj/copy, mob/living/creator, destroy_original = FALSE, no_googlies = FALSE)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_PERMANENTLY_MORTAL, INNATE_TRAIT) // They won't remember their original contents upon ressurection and would just be floating eyes
+	if (no_googlies)
+		overlay_googly_eyes = FALSE
+	CopyObject(copy, creator, destroy_original)
+
+/mob/living/basic/mimic/copy/Life(seconds_per_tick = SSMOBS_DT, times_fired)
+	..()
+	if(idledamage && !ckey && !ai_controller?.blackboard[BB_BASIC_MOB_CURRENT_TARGET]) //Objects eventually revert to normal if no one is around to terrorize
+		adjustBruteLoss(0.5 * seconds_per_tick)
+	for(var/mob/living/victim in contents) //a fix for animated statues from the flesh to stone spell
+		death()
+
+/mob/living/basic/mimic/copy/death()
+	for(var/atom/movable/movable as anything in src)
+		movable.forceMove(get_turf(src))
+	return ..()
+
+/mob/living/basic/mimic/copy/wabbajack(what_to_randomize, change_flags = WABBAJACK)
+	visible_message(span_warning("[src] resists polymorphing into a new creature!"))
+
+/mob/living/basic/mimic/copy/animate_atom_living(mob/living/owner)
+	change_owner(owner)
+
+/mob/living/basic/mimic/copy/proc/change_owner(mob/owner)
+	if(isnull(owner) || creator == owner)
+		return
+	unfriend(creator)
+	befriend(owner)
+
+/mob/living/basic/mimic/copy/proc/check_object(obj/target)
+	return ((isitem(target) || isstructure(target)) && !is_type_in_typecache(target, GLOB.animatable_blacklist))
+
+/mob/living/basic/mimic/copy/proc/CopyObject(obj/original, mob/living/user, destroy_original = FALSE)
+	if(destroy_original || check_object(original))
```

Comment:
Is it correct that destructive mimics bypass the safety checks? Might be worth adding a comment to `check_object` or giving it a more specific proc name if so to explain why.